sdks/java/core/src/main/java/org/apache/beam/sdk/util/ActiveWindowSet.java

```diff
@@ -108,42 +101,40 @@ void onMerge(Collection<W> toBeMerged, Collection<W> activeToBeMerged, W mergeRe
   void persist();
 
   /**
-   * Return the ACTIVE window into which {@code window} has been merged.
-   * Return {@code window} itself if it is ACTIVE. Return null if {@code window} has not
-   * yet been seen.
+   * Return (a view of) the set of currently ACTIVE and NEW windows.
    */
-  @Nullable
-  W mergeResultWindow(W window);
+  Set<W> getActiveAndNewWindows();
 
   /**
-   * Return (a view of) the set of currently ACTIVE windows.
+   * Return {@code true} if {@code window} is ACTIVE.
    */
-  Set<W> getActiveWindows();
+  boolean isActive(W window);
 
   /**
-   * Return {@code true} if {@code window} is ACTIVE.
+   * Return {@code true} if {@code window} is ACTIVE or NEW.
    */
-  boolean isActive(W window);
+  boolean isActiveOrNew(W window);
```

Comment:
This method in the interface is only used for implementations to `checkState`, so it doesn't really need to be here. Or keep symmetry between `ensureWindowExists` and this method, but I'd just delete as much of the public API as you can. Doesn't have to happen in this change, just a thought.